### PR TITLE
[PROGRAMMES-6913] UK Only download button

### PR DESCRIPTION
**Pull request by tomwinnington**
Ticket:
https://jira.dev.bbc.co.uk/browse/PROGRAMMES-6913

Link:
http://sandbox.bbc.co.uk/programmes/p02nrspn/episodes/downloads

screen shot 2019-01-28 at 15 03 27

screen shot 2019-01-28 at 15 06 46

### CODE REVIEW

**Comment by programmes-developers** [comment]
Can one of the admins verify this patch?

File: src/Ds2013/Presenters/Utilities/Download/DownloadPresenter.php [hunk 1]
```diff
@@ -31,7 +31,7 @@ public function __construct(
         Version $version,
         ?Podcast $podcast,
         bool $secondary = true,
-        array $options = []
+        array $options = ['showUkOnly' => true]
```
Comment (cisco87) [inline src/Ds2013/Presenters/Utilities/Download/DownloadPresenter.php]: I think this is not sustainable, could you create a constant or a value for this? 
```
const DEFAULT_OPTIONS = ['showUkOnly' => true];

public function __construct(array $options = []) {
    if (empty($options) {
        $options = self::DEFAULT_OPTIONS;
    }
}
```

I know your solution is easy and simple, but I see it as a time bomb I'm not sure if you agree

File: src/Ds2013/Presenters/Utilities/Download/DownloadPresenter.php [hunk 2]
```diff
@@ -87,4 +87,13 @@ public function getPositionType()
     {
         return ($this->secondary === true) ? 'secondary' : 'subtle';
     }
+
+    public function showUkOnly(): bool
+    {
+        if(array_key_exists('showUkOnly', $this->options) ){
+            return $this->getOption('showUkOnly');
```
Comment (cisco87) [inline src/Ds2013/Presenters/Utilities/Download/DownloadPresenter.php]: this looks like a lot of boilerplate code, can you change getOption to something like

```
public function getOption($key, $default = null) {
   if (array_key_exists($key, $this->options) {
      return $this->options[$key];
   }
   return $default;
}
```

so you can have showUkOnly only return 
```
return $this->getOption('showUkOnly', true);
```

File: src/Ds2013/Presenters/Utilities/Download/DownloadPresenter.php [hunk 3]
```diff
@@ -25,6 +25,10 @@ class DownloadPresenter extends Presenter
     /** @var bool */
     private $secondary;
 
+    protected $options = [
```
Comment (cisco87) [inline src/Ds2013/Presenters/Utilities/Download/DownloadPresenter.php]: Nice!

**Pull request merged**